Source/Applications/SystemCenterNotification/Scripts/TSX/EmailCategory/EmailCategoryPage.tsx

```diff
@@ -50,40 +42,73 @@ const EmailCategoryPage = (props: IProps) => {
     const category = useAppSelector((state) => EmailCategorySlice.Datum(state, parseInt(props.useParams.id)));
     const status: Application.Types.Status = useAppSelector(EmailCategorySlice.Status);
 
-    const [tab, setTab] = React.useState<tab>('settings');
+    const [tab, setTab] = React.useState<Tab>(getTab());
+    const Tabs = [
+        { Label: 'General Settings', Id: 'settings' },
+        { Label: ' Emails', Id: 'emails' },
+    ];
 
     React.useEffect(() => {
         if (status == 'unintiated' || status == 'changed')
             dispatch(EmailCategorySlice.Fetch());
     }, [status]);
 
+    function getTab(): Tab {
+        if (sessionStorage.hasOwnProperty('EmailCategory.Tab'))
+            return JSON.parse(sessionStorage.getItem('EmailCategory.Tab'));
+        else return 'settings';
+    }
```

Comment:
this function should probably be wrapped in a callback with an empty dep array since it doesn't look like it depends on anything external. That being said, I would likely ensure that session storage doesn't interact strangely with that.